docs/content/docs/data-structures/map.md

```diff
@@ -1,10 +1,449 @@
+
+```javascript
+const schema = {
+    name: "example-container",
+    initialObjects: {
+        initalMap: SharedMap,
+    },
+    dynamicObjectTypes: [SharedMap]
+}
+```
+
+Now, you can dynamically create additional `SharedMap` instances and store their handles into the initial map that is always provided in the container.
+
+```javascript
+const schema = {
+    name: "example-container",
+    initialObjects: {
+        initalMap: SharedMap,
+    },
+    dynamicObjectTypes: [SharedMap]
+}
+
+const { fluidContainer, containerServices } = await client.getContainer(/*service config*/, schema);
+
+const initialMap = fluidContainer.initialObjects.initialMap;
+
+const newSharedMap = fluidContainer.create(SharedMap);
+
+initialMap.set("newSharedMapKey", newSharedMap.handle);
+```
+
+Now, anytime you are looking to load the `newSharedMap` back at a later time, all you have to do is load it back from the stored handle.
+
+```javascript
+const newSharedMapHandle = initialMap.get("newSharedMapKey");
+const newSharedMap = await newSharedMapHandle.get();
+```
+
+{{< callout note >}}
+
+Loading any DDS from its handle is an asynchronous operation. Whereas objects defined in `initialObjects` will be available synchronously once the container is provided, you will need to use an `await` or chain Promises on the `handle.get()` call.
+
+{{< /callout >}}
+
+#### Nested shared objects example
+
+Let's take another look at the example from the **Storing objects** section above to see how it can be updated to use nested `SharedMaps`. This will introduce some data hierarchy and avoid having a flat map that has both the tasks and their fields themselves on the same level. To do so, let's start from the earlier model but with two tasks now.
+
+```json
+{
+    "task1": {
+        "title": "Awesome Task",
+        "description": "Doing the most awesome things",
+        "assignedTo": {
+            "name": "Joe Schmo",
+            "email": "joeschmo@email.com",
+            "address": "1234 Fluid Way"
+        }
+    },
+    "task2": {
+        "title": "Even More Awesome Task",
+        "description": "Doing even more awesome things",
+        "assignedTo": {
+            "name": "Jane Doe",
+            "email": "janedoe@email.com",
+            "address": "5678 Framework Street"
+        }
+    }
+}
+```
+Here, you can have each task itself be its own `SharedMap` and have a parent `SharedMap` hold all of the handles to each task.
+
+Then, the initial map would look like:
+```json
+{
+    "task1": task1MapHandle,
+    "task2": task2MapHandle
+}
+```
+And the `task1` map would have:
+```json
+{
+    "title": "Awesome Task",
+    "description": "Doing the most awesome things",
+    "assignedTo": {
+        "name": "Joe Schmo",
+        "email": "joeschmo@email.com",
+        "address": "1234 Fluid Way"
+    }
+}
```

Comment:
```suggestion

To break this apart, you can have each task itself be its own `SharedMap` and have a parent `SharedMap` hold all of the handles to each task. Then the initial map would look like:

```json
{
    "task1": task1MapHandle,
    "task2": task2MapHandle
}
```

And the `task1` map would look like:

```json
{
    "title": "Awesome Task",
    "description": "Doing the most awesome things",
    "assignedTo": {
        "name": "Joe Schmo",
        "email": "joeschmo@email.com",
        "address": "1234 Fluid Way"
    }
}
```